cognite_toolkit/_cdf_tk/utils/table_writers.py

```diff
@@ -0,0 +1,243 @@
+import csv
+import importlib.util
+import json
+from abc import abstractmethod
+from dataclasses import dataclass
+from functools import lru_cache
+from io import TextIOWrapper
+from pathlib import Path
+from types import MappingProxyType
+from typing import IO, TYPE_CHECKING, Any, ClassVar, Generic, Literal, TypeAlias, TypeVar
+
+from cognite_toolkit._cdf_tk.exceptions import ToolkitMissingDependencyError, ToolkitValueError
+from cognite_toolkit._cdf_tk.utils import humanize_collection, to_directory_compatible
+from cognite_toolkit._cdf_tk.utils.file import yaml_safe_dump
+
+if TYPE_CHECKING:
+    import pyarrow as pa
+    import pyarrow.parquet as pq
+
+FileFormat: TypeAlias = Literal["csv", "parquet", "yaml"]
+DataType: TypeAlias = Literal["string", "integer", "float", "boolean", "datetime", "date", "time", "json"]
+Rows: TypeAlias = list[dict[str, Any]]
```

Comment:
is `Any` correct here, or should we use a nested json type?

Reply:
Did a more specific type, however, has to keep the `object` and `list` as we do support `json`.